--- hipcc.pl	2022-08-12 23:24:41.839873569 +0300
+++ hipcc-patched.pl	2022-08-12 23:30:01.991625302 +0300
@@ -105,6 +105,8 @@
 $HIP_VERSION    =   $hipvars::HIP_VERSION;
 $HSA_PATH       =   $hipvars::HSA_PATH;
 $HIP_ROCCLR_HOME =   $hipvars::HIP_ROCCLR_HOME;
+$HIP_ENV_CXX_FLAGS = $hipvars::HIP_ENV_CXX_FLAGS;
+$HIP_ENV_CC_FLAGS = $hipvars::HIP_ENV_CC_FLAGS;
 
 if ($HIP_PLATFORM eq "amd") {
   # If using ROCclr runtime, need to find HIP_ROCCLR_HOME
@@ -188,8 +190,8 @@
     } else {
       $HIPCXXFLAGS .= " -std=c++11";
     }
-    $HIPCXXFLAGS .= " -isystem \"$HIP_CLANG_INCLUDE_PATH/..\"";
-    $HIPCFLAGS .= " -isystem \"$HIP_CLANG_INCLUDE_PATH/..\"";
+    $HIPCXXFLAGS .= " -isystem \"$HIP_CLANG_INCLUDE_PATH/..\" $HIP_ENV_CXX_FLAGS\ ";
+    $HIPCFLAGS .= " -isystem \"$HIP_CLANG_INCLUDE_PATH/..\" $HIP_ENV_CC_FLAGS\ ";
     $HIPLDFLAGS .= " -L\"$HIP_LIB_PATH\"";
     if ($isWindows) {
       $HIPLDFLAGS .= " -lamdhip64";
