--- hipconfig.pl	2022-08-12 23:24:38.582886230 +0300
+++ hipconfig-patched.pl	2022-08-12 23:30:21.291549676 +0300
@@ -47,6 +47,7 @@
 $ROCM_PATH      =   $hipvars::ROCM_PATH;
 $HIP_VERSION    =   $hipvars::HIP_VERSION;
 $HSA_PATH       =   $hipvars::HSA_PATH;
+$HIP_ENV_CC_FLAGS = $hipvars::HIP_ENV_CC_FLAGS;
 
 Getopt::Long::Configure ( qw{bundling no_ignore_case});
 GetOptions(
@@ -81,7 +82,7 @@
     if($isWindows) {
         $CPP_CONFIG .= " -I\"$HIP_PATH_INCLUDE\" -I\"$HIP_CLANG_INCLUDE\"";
     } else {
-        $CPP_CONFIG .= " -I$HIP_PATH_INCLUDE -I$HIP_CLANG_INCLUDE -I$HSA_PATH/include";
+        $CPP_CONFIG .= " -I$HIP_PATH_INCLUDE -I$HIP_CLANG_INCLUDE -I$HSA_PATH/include $HIP_ENV_CC_FLAGS\ ";
     }
 }
 if ($HIP_PLATFORM eq "nvidia") {
